src/Runtime/OMResize.inc

```diff
@@ -0,0 +1,353 @@
+    coeffs_buffer[1] = ratio > 0.5;
+    break;
+  case 1: // round_prefer_ceil
+    coeffs_buffer[0] = ratio < 0.5;
+    coeffs_buffer[1] = ratio >= 0.5;
+    break;
+  case 2: // floor
+    coeffs_buffer[0] = 1;
+    coeffs_buffer[1] = 0;
+    break;
+  case 4: // ceil
+    coeffs_buffer[0] = 0;
+    coeffs_buffer[1] = 1;
+    break;
+  }
+}
+
+static void cubic_coeffs(float ratio, float *coeffs_buffer, int mode) {
+  float A = -0.75;
+  // A may have different value for different coordinate_transformation_mode
+  // Currently, only default mode is supported
+  coeffs_buffer[0] =
+      ((A * (ratio + 1) - 5 * A) * (ratio + 1) + 8 * A) * (ratio + 1) - 4 * A;
+  coeffs_buffer[1] = ((A + 2) * ratio - (A + 3)) * ratio * ratio + 1;
+  coeffs_buffer[2] =
+      ((A + 2) * (1 - ratio) - (A + 3)) * (1 - ratio) * (1 - ratio) + 1;
+  coeffs_buffer[3] =
+      ((A * ((1 - ratio) + 1) - 5 * A) * ((1 - ratio) + 1) + 8 * A) *
+          ((1 - ratio) + 1) -
+      4 * A;
+}
+
+static void get_neighbor(float x, int64_t n, int limit, float *data,
+    float *points, int exclude_outside) {
+
+  // inline the python get_neighbor_idx function without real padding
+  // Avoid malloc/free
+  // nearest indx: identify the central idx first, then select from both side
+  // If the central idx is right to x (>= x), favor the left one
+  // == from the example
+
+  int pad_width = ceil(n / 2);
+  x += pad_width;
+  float r = x - floor(x);
+
+  int start, end;
+  int c;
+  if (r > 0.5) {
+    c = (int)(floor(x)) + 1;
+  } else {
+    c = (int)(floor(x));
+  }
+
+  int rest = n - 1;
+  int half = rest / 2;
+  if (rest == 0) {
+    start = end = c;
+  } else if (rest % 2 == 0) {
+    start = c - half;
+    end = c + half;
+  } else if (r == 0) {
+    start = c - half - 1;
+    end = c + half;
+  } else {
+    if (r > 0.5) {
+      end = c + half;
+      start = c - half - 1;
+    } else {
+      end = c + half + 1;
+      start = c - half;
+    }
+  }
+
+  start -= pad_width;
+  end -= pad_width;
+
+  for (int i = start; i <= end; i++) {
+    if (i < 0) {
+      if (exclude_outside)
+        points[i - start] = 0;
+      else
+        points[i - start] = data[0];
+    } else if (i >= limit) {
+      if (exclude_outside)
+        points[i - start] = 0;
+      else
+        points[i - start] = data[limit - 1];
+    } else {
+      points[i - start] = data[i];
+    }
+  }
+}
+
+typedef void (*Coeff_Func_t)(float, float *, int mode);
+
+static float interpolate_1d_with_x(OMTensor *data, float scale_factor, float x,
+    Coeff_Func_t get_coeffs, float *coeffs_buffer, int coeffs_n, float roi,
+    float extrapolation_value, int coordinate_transformation_mode,
+    int exclude_outside, int mode) {
+
+  int64_t input_width = omTensorGetShape(data)[0];
+  float x_ori;
+  switch (coordinate_transformation_mode) {
+  case 0: // half_pixel
+    x_ori = (x + 0.5) / scale_factor - 0.5;
+    break;
+  case 1: // asymmetric
+    x_ori = x / scale_factor;
+    break;
+  }
```

Comment:
[Suggestion]: could add a default case and assert that coordinate_transformation_mode can never be any value other than zero or one.